src/Dialect/ONNX/ElementsAttr/DisposablePool.cpp

```diff
@@ -114,17 +114,18 @@ void DisposablePool::scrub(ModuleOp moduleOp, OpAttrDictionary opsAttrs) {
       return llvm::make_range(batchBegin, batchEnd);
     };
     // Parallel worker body: Fetch and process batches until there are no more.
-    auto work = [&fetchBatch](size_t threadNumber) {
+    auto work = [&fetchBatch, &translationMutex](size_t threadNumber) {
       for (;;) {
         auto batch = fetchBatch();
         if (batch.empty())
           break;
         for (auto &[id, translation] : batch) {
           auto &[disposable, dense] = translation;
           dense = disposable.toDenseElementsAttr();
-          // TODO: Consider calling disposable.dispose() here to free up memory
-          //       on the go to make memory available to create the next
-          //       DenseElementsAttr. In that case should we lock mutex?
+          {
```

Comment:
This `{}` block looks redundant, since the scope of mutex can be the scope of for-loop.

Reply:
I thought we need to use mutex only for `disposable.dispose()` so that `dense = disposable.toDenseElementsAttr();` runs in parallel.

Reply:
It’s Ok, but be careful when creating attributes runs in parallel, we may have a spike again before the buffers are disposed.